### PR TITLE
MULE-11280: Policy parametrization must be used to create policy context

### PR DESCRIPTION
**Pull request by pablokraan**

### CODE REVIEW

File: modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/policy/DefaultApplicationPolicyInstance.java [hunk 1]
```diff
@@ -66,6 +67,7 @@ private void initPolicyContext() {
     if (policyContext == null) {
       ArtifactContextBuilder artifactBuilder =
           newBuilder().setArtifactType(APP)
+              .setArtifactProperties(new HashMap<>(parametrization.getParameters()))
```
Comment (elrodro83) [inline modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/policy/DefaultApplicationPolicyInstance.java]: why not Collections.immutableMap()?

File: modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/DeploymentServiceTestCase.java [hunk 2]
```diff
@@ -3006,7 +3010,7 @@ public void synchronizesDomainRedeployFromClient() throws Exception {
 
   @Test
   public void appliesApplicationPolicy() throws Exception {
-    doApplicationPolicyExecutionTest(parameters -> true, 1);
+    doApplicationPolicyExecutionTest(parameters -> true, 1, "policyPropertyValue");
```
Comment (elrodro83) [inline modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/DeploymentServiceTestCase.java]: extract constant

File: modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/DeploymentServiceTestCase.java [hunk 3]
```diff
@@ -3031,24 +3035,28 @@ public void appliesMultipleApplicationPolicies() throws Exception {
...
     policyManager.registerPolicyTemplate(fooPolicyFileBuilder.getArtifactFile());
 
     ApplicationFileBuilder applicationFileBuilder = createExtensionApplicationWithServices();
     addPackedAppFromBuilder(applicationFileBuilder);
 
 
     policyManager.addPolicy(applicationFileBuilder.getId(), fooPolicyFileBuilder.getId(),
-                            new PolicyParametrization(FOO_POLICY_ID, pointcut, 1, emptyMap()));
+                            new PolicyParametrization(FOO_POLICY_ID, pointcut, 1,
+                                                      singletonMap("policyPropertyKey", "policyPropertyValue")));
```
Comment (marianogonzalez) [inline modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/DeploymentServiceTestCase.java]: constants, at least for the value

File: modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/policy/DefaultApplicationPolicyInstance.java (hunk 1 above)
Comment (pablokraan) [inline modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/policy/DefaultApplicationPolicyInstance.java]: the original is immutable, later during the deployment the deployer adds more properties there

**Pull request merged**